Flux inference error on ascend npu

### Describe the bug

It fails to run the demo flux inference code. reporting errors:
> RuntimeError: call aclnnRepeatInterleaveIntWithDim failed, detail:EZ1001: [PID: 23975] 2025-01-02-11:00:00.313.502 self not implemented for DT_DOUBLE, should be in dtype support list [DT_UINT8,DT_INT8,DT_INT16,DT_INT32,DT_INT64,DT_BOOL,DT_FLOAT16,DT_FLOAT,DT_BFLOAT16,].

### Reproduction

```python
import torch
try:
    import torch_npu  # type: ignore # noqa
    from torch_npu.contrib import transfer_to_npu  # type: ignore # noqa
    is_npu = True
except ImportError:
    print("torch_npu not found")
    is_npu = False
from diffusers import FluxPipeline

pipe = FluxPipeline.from_pretrained(
        "black-forest-labs/FLUX.1-dev",
        torch_dtype=torch.bfloat16)
pipe.to('cuda')

prompt = "A cat holding a sign that says hello world"
image = pipe(
    prompt,
    height=1024,
    width=1024,
    guidance_scale=3.5,
    num_inference_steps=50,
    max_sequence_length=512,
    generator=torch.Generator("cpu").manual_seed(0)
).images[0]
image.save("flux-dev.png")
```

### Logs

```shell
Traceback (most recent call last):
  File "/home/pagoda/exp.py", line 18, in <module>
    image = pipe(
  File "/usr/local/python3.10.13/lib/python3.10/site-packages/torch/utils/_contextlib.py", line 116, in decorate_context
    return func(*args, **kwargs)
  File "/usr/local/python3.10.13/lib/python3.10/site-packages/diffusers/pipelines/flux/pipeline_flux.py", line 889, in __call__
    noise_pred = self.transformer(
  File "/usr/local/python3.10.13/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1553, in _wrapped_call_impl
    return self._call_impl(*args, **kwargs)
  File "/usr/local/python3.10.13/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1562, in _call_impl
    return forward_call(*args, **kwargs)
  File "/usr/local/python3.10.13/lib/python3.10/site-packages/diffusers/models/transformers/transformer_flux.py", line 492, in forward
    image_rotary_emb = self.pos_embed(ids)
  File "/usr/local/python3.10.13/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1553, in _wrapped_call_impl
    return self._call_impl(*args, **kwargs)
  File "/usr/local/python3.10.13/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1562, in _call_impl
    return forward_call(*args, **kwargs)
  File "/usr/local/python3.10.13/lib/python3.10/site-packages/diffusers/models/embeddings.py", line 1253, in forward
    cos, sin = get_1d_rotary_pos_embed(
  File "/usr/local/python3.10.13/lib/python3.10/site-packages/diffusers/models/embeddings.py", line 1157, in get_1d_rotary_pos_embed
    freqs_cos = freqs.cos().repeat_interleave(2, dim=1).float()  # [S, D]
RuntimeError: call aclnnRepeatInterleaveIntWithDim failed, detail:EZ1001: [PID: 23975] 2025-01-02-11:00:00.313.502 self not implemented for DT_DOUBLE, should be in dtype support list [DT_UINT8,DT_INT8,DT_INT16,DT_INT32,DT_INT64,DT_BOOL,DT_FLOAT16,DT_FLOAT,DT_BFLOAT16,].

[ERROR] 2025-01-02-11:00:00 (PID:23975, Device:0, RankID:-1) ERR01100 OPS call acl api failed
```
```


### System Info

- 🤗 Diffusers version: 0.32.
- Platform: Linux-5.10.0-136.36.0.112.4.oe2203sp1.x86_64-x86_64-with-glibc2.35
- Running on Google Colab?: No
- Python version: 3.10.13
- PyTorch version (GPU?): 2.4.0+cpu (False)
- Flax version (CPU?/GPU?/TPU?): not installed (NA)
- Jax version: not installed
- JaxLib version: not installed
- Huggingface_hub version: 0.27.0
- Transformers version: 4.46.3
- Accelerate version: 1.1.0
- PEFT version: 0.13.2
- Bitsandbytes version: not installed
- Safetensors version: 0.4.5
- xFormers version: not installed
- Accelerator: NA
- Using GPU in script?: Ascend 910B
- Using distributed or parallel set-up in script?: <fill in>

### Who can help?

_No response_

--- a/src/diffusers/models/embeddings.py
+++ b/src/diffusers/models/embeddings.py
@@ -1248,7 +1248,8 @@ class FluxPosEmbed(nn.Module):
         sin_out = []
         pos = ids.float()
         is_mps = ids.device.type == "mps"
-        freqs_dtype = torch.float32 if is_mps else torch.float64
+        is_npu = ids.device.type == "npu"
+        freqs_dtype = torch.float32 if (is_mps or is_npu) else torch.float64
         for i in range(n_axes):
             cos, sin = get_1d_rotary_pos_embed(
                 self.axes_dim[i],

